tests/test_expression.py

```diff
@@ -44,13 +45,24 @@
     ("0 | 1", 1),
     ("1 | 1", 1),
     ("1 | 2", 3),
-    # This type of expression is not supported by the parser and will fail.
-    # ('4 * 1 + 1', 5),
+    ("1 | 2 | 4", 7),
+    ("1 & 1 * 4", 0),
+    ("(1 & 1) * 4", 4),
+    ("4 * 1 + 1", 5),
     ("-42", -42),
     ("42 + (-42)", 0),
     ("A + 5", 13),
     ("21 - B", 8),
     ("A + B", 21),
+    ("~1", -2),
```

Comment:
```suggestion
    ("0o77 + 0o66", 117),
    ("~1", -2),
```

Maybe add some cases for octal?